src/PrestaShopBundle/Resources/config/routing/admin/sell/orders/carts.yml

```diff
@@ -68,20 +68,44 @@ admin_carts_set_free_shipping:
   options:
     expose: true
 
-admin_carts_add_rule:
-  path: /{cartId}/rules/add
+admin_carts_add_cart_rule:
+  path: /{cartId}/cart-rules
   methods: [POST]
   defaults:
     _controller: PrestaShopBundle:Admin/Sell/Order/Cart:addCartRuleToCart
     _legacy_controller: AdminCarts
   options:
     expose: true
 
-admin_carts_delete_rule:
-  path: /{cartId}/rules/{cartRuleId}/delete
+admin_carts_delete_cart_rule:
+  path: /{cartId}/cart-rules/{cartRuleId}/delete
   methods: [POST]
   defaults:
     _controller: PrestaShopBundle:Admin/Sell/Order/Cart:removeCartRuleFromCart
     _legacy_controller: AdminCarts
   options:
     expose: true
+
+admin_carts_add_product:
+  path: /{cartId}/products
+  methods: [POST]
+  defaults:
+    _controller: PrestaShopBundle:Admin/Sell/Order/Cart:addProduct
+    _legacy_controller: AdminCarts
+  options:
+    expose: true
```

Comment:
Just a question for my personal knowledge ^^, this `exposed` option is used for fos_js_routing?

Reply:
yes :)

Reply:
yes